api/token.go

```diff
@@ -472,13 +472,24 @@ func (a *API) IdTokenGrant(ctx context.Context, w http.ResponseWriter, r *http.R
 			}
 		}
 
+		// an account with a previously unconfirmed email + password
+		// combination or phone may exist. so now that there is an
+		// OAuth identity bound to this user, and since they have not
+		// confirmed their email or phone, they are unaware that a
+		// potentially malicious door exists into their account; thus
+		// the password and phone needs to be removed.
+		if terr = user.RemoveUnconfirmedIdentities(tx); terr != nil {
+			return internalServerError("Error updating user").WithInternalError(terr)
+		}
+
```

Comment:
hmm this isn't where the oauth flow takes place though - this should be done in the `/callback` endpoint [here](https://github.com/supabase/gotrue/blob/master/api/external.go#L106)

Reply:
Absolutely right. I've accidentally mistaken it for this as the code looks eerily similar in both places. It's been moved now.